src/main/resources/Confluence/Macros/DetailsSummary.xml

```diff
@@ -0,0 +1,420 @@
+     contentPlain = services.rendering.render(block2.getChildren()[0], 'plain/1.0')
+    }
+    if (content1.trim()!="" &amp;&amp; block2!=null) {
+      result[content1.trim()] = [ text : contentPlain, wiki: contentWiki];
+      findInDropDownAndLinks(block2, result[content1.trim()])
+    }
+  }
+}
+
+def findInXDOM(xdom, result) {
+  def blocks = xdom.getBlocks(new ClassBlockMatcher(MacroBlock.class), Block.Axes.DESCENDANT);
+  for (def block in blocks) {
+    def content = block.getContent()
+    if (content!=null &amp;&amp; content!="") {
+      def newxdom = services.rendering.parse(content, 'xwiki/2.1')
+      findInXDOM(newxdom, result);
+      if (block.getId()=="details") {
+        addDebug("Found details macro block")
+        def details_id = block.getParameter("id")
+        if (details_id == null || details_id == '') {
+          details_id = block.hashCode()
+        }
+        result[details_id] = [:];
+        findInXDOMDetails(newxdom, result[details_id])
+      }
+    }
+  }
+}
+
+def findData(page) {
+  def result = [:]
+  addDebug("Checking page " + page)
+  def pageDoc = xwiki.getDocument(page)
+  def xdom = pageDoc.getXDOM()
+  findInXDOM(xdom, result)
+  return result;
+}
+
+// initialize column set needed for livetable configuration
+Set columns = ["Title"]
+
+// use the full document name without '.WebHome' to find subpages
+def fullName = doc.getFullName()
+def webHomeIndex = fullName.lastIndexOf(".WebHome")
+if (webHomeIndex != -1) {
+  fullName = fullName.substring(0, webHomeIndex)
+}
+subPages = services.query.xwql("where doc.space like :currentDocPre order by trim(trailing '.WebHome' from doc.fullName)").bindValue('currentDocPre', fullName + '.%').execute()
+
+// add the values from the first column of every details macro to the livetable columns set
+for (page in subPages) {
+  if (page != null) {
+    def detailsData = findData(page)
+    for (detailsMacro in detailsData) {
+      for (pair in detailsMacro.getValue()) {
+        columns += [pair.getKey().trim().replace(':', '')]
+      }
+    }
+  }
+}
+
+// store variable in the xcontext so we can use it in velocity
+xcontext.put("columns", columns)
+{{/groovy}}
+
+{{velocity}}
+## get the variable back from the context
+#set ($columns = $xcontext.get("columns"))
+## create the column configuration - currently all columns are considered text columns
+#set ($columnsProperties = {})
+#foreach ($column in $columns)
+  #set ($discard = $columnsProperties.put($column, { "type" : "none", "html" : "true" }))
+  ##set ($discard = $columnsProperties.put($column, { "type" : "text" }))
+#end
+## set the page to query for results and pass the page from which the macro is called as an extra parameter called 'source'
+#set ($options = {
+  "resultPage":"Confluence.Macros.DetailsSummaryResult",
+  "extraParams":"source=$escapetool.url($doc)"
+})
+#livetable("testtable" $columns $columnsProperties $options)
```

Comment:
I'd appreciate a more explicit livetable ID, like "detailssummary". What's more, we may need to ensure uniqueness, maybe by appending some counter. Counter that could be put in the context.